### ISSUE TITLE
Flaky test "OGR::DataSource#sync_to_disk"

### ISSUE THREAD

**Issue by oleksii-leonov** [telus-agcg/ffi-gdal#83]
Flaky test (randomly fails from time to time):

```
rspec ./spec/unit/ogr/data_source_spec.rb:218 # OGR::DataSource#sync_to_disk 

Failures:

  1) OGR::DataSource#sync_to_disk 
     Failure/Error:
       OGR::ErrorHandling.handle_ogr_err("Unable to syn datasource to disk") do
         FFI::OGR::API.OGR_DS_SyncToDisk(@c_pointer)
       end

     OGR::Failure:
       Unable to syn datasource to disk
     # ./lib/ogr/data_source.rb:239:in `sync_to_disk'
     # ./spec/unit/ogr/data_source_spec.rb:219:in `block (3 levels) in <top (required)>'
```

**Comment by oleksii-leonov** [comment]
The minimum case to reproduce:

```ruby
bundle exec rspec ./spec/unit/gdal/geo_transform_spec.rb[1:1:2:1] ./spec/unit/ogr/data_source_spec.rb[1:13:1] --seed 1
```

```
Randomized with seed 1
.F

Failures:

  1) OGR::DataSource#sync_to_disk 
     Failure/Error:
       OGR::ErrorHandling.handle_ogr_err("Unable to syn datasource to disk") do
         FFI::OGR::API.OGR_DS_SyncToDisk(@c_pointer)
       end
     
     OGR::Failure:
       Unable to syn datasource to disk
     # ./lib/ogr/data_source.rb:239:in `sync_to_disk'
     # ./spec/unit/ogr/data_source_spec.rb:219:in `block (3 levels) in <top (required)>'

Failed examples:

rspec ./spec/unit/ogr/data_source_spec.rb:218 # OGR::DataSource#sync_to_disk 
```

**Comment by oleksii-leonov** [comment]
Low-level code to reproduce:

This one works fine:
```ruby
driver_ptr = FFI::OGR::API.OGRGetDriverByName("Memory")
data_source_ptr = FFI::OGR::API.OGR_Dr_CreateDataSource(driver_ptr, "spec123", nil)

OGR::ErrorHandling.handle_ogr_err("Unable to syn datasource to disk") do
  FFI::OGR::API.OGR_DS_SyncToDisk(data_source_ptr)
end
```

This one fails:
```ruby
begin
  FFI::GDAL::GDAL.GDALLoadWorldFile("test123", FFI::MemoryPointer.new(:double, 6))
rescue GDAL::OpenFailure => e
  puts e.message
end

driver_ptr = FFI::OGR::API.OGRGetDriverByName("Memory")
data_source_ptr = FFI::OGR::API.OGR_Dr_CreateDataSource(driver_ptr, "spec123", nil)

OGR::ErrorHandling.handle_ogr_err("Unable to syn datasource to disk") do
  FFI::OGR::API.OGR_DS_SyncToDisk(data_source_ptr)
end
```

**Comment by oleksii-leonov** [comment]
In general, it looks strange.

However, I believe doing SyncToDisk for the Memory dataset is not typical, and the expected result is not clear to me (e.g. how it's possible to flush disk cache for an in-memory dataset that has no file open).

My proposal is to use a file-based dataset for this particular spec.

For example, the following code works fine:

```ruby
begin
  FFI::GDAL::GDAL.GDALLoadWorldFile("test123", FFI::MemoryPointer.new(:double, 6))
rescue GDAL::OpenFailure => e
  puts e.message
end

driver_ptr = FFI::OGR::API.OGRGetDriverByName("ESRI Shapefile")
data_source_ptr = FFI::OGR::API.OGR_Dr_CreateDataSource(driver_ptr, "spec123", nil)
FFI::OGR::API.OGR_DS_SyncToDisk(data_source_ptr)

# OGRERR_NONE
```

**Comment by oleksii-leonov** [comment]
PR with fix: #90.
